test/distrib/bazel/test_single_bazel_version.sh

```diff
@@ -51,12 +51,31 @@ FAILED_TESTS=""
 export OVERRIDE_BAZEL_VERSION="$VERSION"
 # when running under bazel docker image, the workspace is read only.
 export OVERRIDE_BAZEL_WRAPPER_DOWNLOAD_DIR=/tmp
-bazel build -- //... "${EXCLUDED_TARGETS[@]}" || FAILED_TESTS="${FAILED_TESTS}Build "
+
+# pick the Bazel wrapper
+BAZEL="$(pwd)/tools/bazel"
+
+# validate the Bazel version
+if ! ($BAZEL version | grep -q "${VERSION}"); then
+  echo "Incorrect Bazel version! Want=${VERSION} Seen=$($BAZEL version)"
+  exit 1
+fi
+
+# clean the caches and downloaded archives to make sure we build from scratch
+$BAZEL clean --expunge
```

Comment:
Requiring "bazel clean" seems speculative. Do we really have a clear evidence that this is necessary?
I'm pretty sure bazel is not stupid and it recognizes that if you change the bazel version, a full rebuild is needed.